src/compiler/emitter.ts

```diff
@@ -186,9 +186,9 @@ namespace ts {
             for (const inputFileName of configFile.fileNames) {
                 if (fileExtensionIs(inputFileName, Extension.Dts)) continue;
                 const js = getOutputJSFileName(inputFileName, configFile, ignoreCase);
-                addOutput(js);
+                if (!configFile.options.emitDeclarationOnly) addOutput(js);

```

Comment:
This is not needed... `getOutputJSFileName` handles this. If you add below test case [here](https://github.com/microsoft/TypeScript/blob/master/src/testRunner/unittests/tsbuild/emitDeclarationOnly.ts#L108), you will see it will pass

```ts
it("verify there is nothing to build after being built", () => {
            const fs = projFs.shadow();
            const host = new fakes.SolutionBuilderHost(fs);
            let builder = createSolutionBuilder(host, ["/src"], { verbose: true });
            builder.build();
            host.clearDiagnostics();
            tick();
            builder = createSolutionBuilder(host, ["/src"], { verbose: true });
            builder.build();
            host.assertDiagnosticMessages(
                getExpectedDiagnosticForProjectsInBuild("src/tsconfig.json"),
                [Diagnostics.Project_0_is_up_to_date_because_newest_input_1_is_older_than_oldest_output_2, "src/tsconfig.json", "src/src/a.ts", "src/lib/a.d.ts"],
            );
        });
```

Reply:
Should the builds be shorter though? I can set up a repro